packages/vscode-ui5-language-assistant/src/registerManifestSchema.ts

```diff
@@ -0,0 +1,104 @@
+/* istanbul ignore file - temporary workaround */
+import { isArray, find, some, clone, findIndex } from "lodash";
+import { workspace } from "vscode";
+import {
+  getManifestSchemaConfigEntry,
+  IJSONValidationExtensionPoint,
+  MANIFEST_COMMENT,
+  MANIFEST_FILE_MATCH,
+} from "./manifestSchemaConfig";
+
+/**
+ *  The configuration will be written in workspace.settings
+ */
+export function registerManifestSchema(): void {
+  const jsonSchemaConfig:
+    | IJSONValidationExtensionPoint
+    | undefined = workspace.getConfiguration().get("json.schemas");
+  if (isArray(jsonSchemaConfig)) {
+    const previousConfigEntryIndex = getIndexOfPreviousConfigEntry(
+      jsonSchemaConfig
+    );
+    if (previousConfigEntryIndex !== -1) {
+      updatePreviousConfigEntry(jsonSchemaConfig, previousConfigEntryIndex);
+    } else if (isMissingConfig(jsonSchemaConfig)) {
+      createConfig(jsonSchemaConfig);
+    } else {
+      // No change is needed
+    }
+  }
+}
+
+function isMissingConfig(
+  jsonSchemaConfig: IJSONValidationExtensionPoint[]
+): boolean {
+  const missingConfigEntry =
+    find(
+      jsonSchemaConfig,
+      (jsonSchemaConfigEntry: IJSONValidationExtensionPoint) => {
+        if (isArray(jsonSchemaConfigEntry.fileMatch)) {
+          return some(
+            jsonSchemaConfigEntry.fileMatch,
+            (fileMatchEntry) => fileMatchEntry === MANIFEST_FILE_MATCH
+          );
+        }
+
+        return jsonSchemaConfigEntry.fileMatch === MANIFEST_FILE_MATCH;
```

Comment:
You mentioned that the fileMatch property does not work when it is not an array.
So this code of dealing with both cases may be redundant.

However we saw some VSCode API in which it **could** be an single string.
Perhaps it does not matter as this is a temp workaround.
I will try to find more info.